### ISSUE TITLE
KeyNotFoundException 

### ISSUE THREAD

**Issue by jbialobr** [bergerjac/gitextensions#24]
System.Collections.Generic.KeyNotFoundException was unhandled by user code
  Message=Dany klucz nie był obecny w słowniku.
  Source=mscorlib
  StackTrace:
       w System.Collections.Generic.Dictionary`2.get_Item(TKey key)
       w GitCommands.Git.RemoteInfo..ctor(String remoteShowOutput) w D:\projekty\gitextensions\GitCommands\Git\RemoteInfo.cs:wiersz 189
       w GitCommands.GitModule.<GetRemotesInfo>b__2a(String remote) w D:\projekty\gitextensions\GitCommands\Git\GitModule.cs:wiersz 2106
       w System.Linq.Enumerable.<>c__DisplayClass12`3.<CombineSelectors>b__11(TSource x)
       w System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
       w System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
       w System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
       w GitUI.UserControls.RepoObjectsTree.<OnUICommandsSourceChanged>b__e() w D:\projekty\gitextensions\GitUI\RepoObjectsTree\RepoObjectsTree.cs:wiersz 75
       w GitUI.UserControls.ValueWatcher`1.<CheckUpdateAsync>b__4() w D:\projekty\gitextensions\GitUI\RepoObjectsTree\ValueWatcher.cs:wiersz 55
       w System.Threading.Tasks.Task`1.InvokeFuture(Object futureAsObj)
       w System.Threading.Tasks.Task.InnerInvoke()
       w System.Threading.Tasks.Task.Execute()
  InnerException:

**Comment by bergerjac** [comment]
If you can reproduce this consistently, can you put a breakpoint at:
`GitCommands\Git\RemoteInfo.cs, line 188`
Right-click on the breakpoint icon, [Condition...]
`!NameToBranch.ContainsKey(remoteBranch)`

![GitEx RemoteInfo BreakpointCondition](https://f.cloud.github.com/assets/1484018/141648/07b0b820-725a-11e2-9fba-54e091bf258c.png)

Then, if you can fix it, that'd be fantastic; otherwise, please comment with the offending values.

**Comment by jbialobr** [comment]
I send you debug data via email.

**Comment by bergerjac** [comment]
@jbialobr can you check [this](http://stackoverflow.com/questions/15125151/git-remote-show-origin-anomaly#comment21288046_15125151) and either respond on stackoverflow (if you have an account) or this one?

**Comment by bergerjac** [comment]
@jbialobr can you also post the lines for `[remote "origin"]`. Also, you may want to delete the answer you posted on the other site.  (Answers are reserved for solving the question/problem.)

**Comment by jbialobr** [comment]
```
[remote "origin"]
    fetch = +refs/heads/*:refs/remotes/origin/*
```

> Also, you may want to delete the answer you posted on the other site. (Answers are reserved for solving the question/problem.)

Where do I have to click to add a comment to question?

**Comment by bergerjac** [comment]
Right below Pascal's comment there's an _Add Comment_ link that expands to a textbox where you can place a comment.

**Comment by jbialobr** [comment]
I can't see it:

![tmp](https://f.cloud.github.com/assets/1003909/206628/1be07fd4-81d6-11e2-9609-55344c68504a.png)

**Comment by bergerjac** [comment]
Apparently, you have to have 50 reputation points before being able to comment.  I'll post an edit to the question.

On the _remote repository_, is the branch named `origin/DowSzkDoZgl` or just `DowSzkDoZgl`?

**Comment by jbialobr** [comment]
`origin/DowSzkDoZgl`

**Comment by bergerjac** [comment]
Is the issue no longer taking place?

Also, what does it show when you perform:  `git ls-remote --heads origin`?

**Comment by jbialobr** [comment]
```
$ git ls-remote --heads origin | grep DowSzkDoZgl
aceb95fa2cf80bfec65272acf0898a2a47d8fac9        refs/heads/DowSzkDoZgl
aceb95fa2cf80bfec65272acf0898a2a47d8fac9        refs/heads/origin/DowSzkDoZgl
```

Yes, this issue is fixed, I closed it.

**Comment by bergerjac** [comment]
Are you sure it was fixed in code?  Or did you fix it by something you did to your repo?

**Comment by jbialobr** [comment]
Yes, it was fixed by e16c10b7f9fb810441038a5f291caa392c58f171
